include/envoy/network/connection.h

```diff
@@ -166,6 +166,30 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    */
   virtual const Network::Address::InstanceConstSharedPtr& remoteAddress() const PURE;
 
+  /**
+   * Credentials of the peer of a socket as decided by SO_PEERCRED.
+   */
+  struct PeerCredentials {
```

Comment:
Should we call this `UnixDomainSocketPeerCredentials` or something like that? FWIW this is not what I thought it was when I read the PR title.

Reply:
Yeah that makes sense to me